src/filter/mod.rs

```diff
@@ -354,15 +354,15 @@ fn parse_spec(spec: &str) -> (Vec<Directive>, Option<inner::Filter>) {
         }
     });
 
-    let filter = filter.map_or(None, |filter| match inner::Filter::new(filter) {
+    let filter = filter.and_then(|filter| match inner::Filter::new(filter) {
```

Comment:
`and_then` returns None if the Option contains None, otherwise it returns T. Which is the same as the `map_or` function call.